### PR TITLE
Add the hyperlink of the transfomer doc

### LINKED ISSUE #120488: Better documentation on _mask and _key_padding_mask in transformers modules
### 📚 The doc issue

The [docs](https://pytorch.org/docs/stable/generated/torch.nn.TransformerEncoder.html) here do not clearly indicate the shapes of the masks and their equivalent types that the transformer modules are expecting.

The only place where one can find shapes and types of expected tensors are only in the [source code docs](https://github.com/pytorch/pytorch/blob/54bac042e711f08e909dda939f70d6f272dbe8ce/torch/csrc/api/include/torch/nn/modules/transformer.h#L45) atm.

### Suggest a potential alternative/fix

A couple of examples of different use cases would really help users understand what is expected what the actual effect is of each mask. 

Would the transformer module operate on both of of them?

Also, a clarification on why one uses a boolean mask for `_key_padding_mask` with `True` values indicating there's padding there and `False` there's no padding which is actually the opposite of other frameworks.

For instance, here's an excerpt from hugging face explaining the same thing but they use 1s, i.e., `True` indicating where the model needs to attend to and 0s `False` where there's padding.

```
Attention masks
Attention masks are tensors with the exact same shape as the input IDs tensor, filled with 0s and 1s.
1s indicate the corresponding tokens should be attended to, 
and 0s indicate the corresponding tokens should not be attended to 
(i.e., they should be ignored by the attention layers of the model).
```

cc @svekars @brycebortree @jbschlosser @bhosmer @cpuhrsch @erichan1 @drisspg @mikaylagawarecki

### PR DESCRIPTION
**Pull request by lancerts**
Fixes #120488 

- The shape for forward pass is clearly stated in the main [transformer class](https://pytorch.org/docs/stable/generated/torch.nn.Transformer.html)

- Boolean mask for _key_padding_mask is also explained in the main transformer class.

Therefore, add the hyperlink to the transformer class explicitly so the user can refer back to the main class. Also, correct several symbols in the transform doc from normal text style to math style.

### CODE REVIEW

File: torch/nn/modules/transformer.py [hunk 1]
```diff
@@ -314,7 +314,7 @@ def forward(
                 compatibility.
 
         Shape:
-            see the docs in Transformer class.
+            see the docs in `Transformer <https://pytorch.org/docs/stable/generated/torch.nn.Transformer.html>` class.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/transformer.py:317]: Could you try this please (similarly for below)

```suggestion
            see the docs in :class:`~torch.nn.Transformer`
```

Comment (lancerts) [reply torch/nn/modules/transformer.py]: Modified all places accordingly.

**Review by mikaylagawarecki — APPROVED** [review]
Thanks a lot!

re the mask discrepancy mentioned on the initial issue 

> Also, a clarification on why one uses a boolean mask for _key_padding_mask with True values indicating there's padding there and False there's no padding which is actually the opposite of other frameworks.

As you mentioned, the mask definition is documented in the Transformer docs.
However, would you be willing to add a `..note::` in the Transformer class that for a boolean mask, `True` means that a value is not included in the attention which is the opposite of what it means in `torch.nn.functional.scaled_dot_product_attention` please?

(If it is easier for you feel free to do this in a follow up)

**Comment by lancerts** [comment]
> Thanks a lot!
> 
> re the mask discrepancy mentioned on the initial issue
> 
> > Also, a clarification on why one uses a boolean mask for _key_padding_mask with True values indicating there's padding there and False there's no padding which is actually the opposite of other frameworks.
> 
> As you mentioned, the mask definition is documented in the Transformer docs. However, would you be willing to add a `..note::` in the Transformer class that for a boolean mask, `True` means that a value is not included in the attention which is the opposite of what it means in `torch.nn.functional.scaled_dot_product_attention` please?
> 
> (If it is easier for you feel free to do this in a follow up)

Actually, i checked the doc in the transfomer to see if Transformer and torch.nn.functional.scaled_dot_product_attention are consistent.
Screenshot 2024-02-26 at 11 55 31 AM

Based on 
```
[src/tgt/memory]_key_padding_mask provides specified elements in the key `to be ignored` by the attention. If a BoolTensor is provided, the positions with the value of True will be ignored while the position with the value of False will be unchanged.
```

**Comment by lancerts** [comment]
> Thanks a lot!
> 
> re the mask discrepancy mentioned on the initial issue
> 
> > Also, a clarification on why one uses a boolean mask for _key_padding_mask with True values indicating there's padding there and False there's no padding which is actually the opposite of other frameworks.
> 
> As you mentioned, the mask definition is documented in the Transformer docs. However, would you be willing to add a `..note::` in the Transformer class that for a boolean mask, `True` means that a value is not included in the attention which is the opposite of what it means in `torch.nn.functional.scaled_dot_product_attention` please?
> 
> (If it is easier for you feel free to do this in a follow up)

Will do.
